micrometer-spring-legacy/src/main/java/io/micrometer/spring/advice/TimedMethodInterceptor.java

```diff
@@ -0,0 +1,87 @@
+            Tags.of("class", invocation.getMethod().getDeclaringClass().getSimpleName(),
+                    "method", invocation.getMethod().getName())
+        );
+    }
+
+    public TimedMethodInterceptor(MeterRegistry registry, TimedTagsResolver timedTagsResolver) {
+        this(registry, (metricName, invocation) -> metricName, timedTagsResolver);
+    }
+
+    public TimedMethodInterceptor(MeterRegistry registry, TimedMetricNameResolver timedMetricNameResolver, TimedTagsResolver timedTagsResolver) {
+        this.registry = registry;
+        this.timedMetricNameResolver = timedMetricNameResolver;
+        this.timedTagsResolver = timedTagsResolver;
+    }
+
+    @Override
+    public Object invoke(MethodInvocation invocation) throws Throwable {
+        Method method = invocation.getMethod();
+        Timed methodLevelTimed = AnnotatedElementUtils.findMergedAnnotation(method, Timed.class);
+        Timed classLevelTimed = AnnotatedElementUtils.findMergedAnnotation(method.getDeclaringClass(), Timed.class);
+        if (methodLevelTimed == null) {
+            method = ReflectionUtils.findMethod(invocation.getThis().getClass(), method.getName(), method.getParameterTypes());
+            methodLevelTimed = AnnotatedElementUtils.findMergedAnnotation(method, Timed.class);
+        }
+        if (classLevelTimed == null && methodLevelTimed == null) {
+            return invocation.proceed();
+        }
+
+        Timer.Sample sample = Timer.start(registry);
+        final String metricName;
+        if (methodLevelTimed != null && !methodLevelTimed.value().isEmpty()) {
+            metricName = methodLevelTimed.value();
+        }
+        else if (classLevelTimed != null && !classLevelTimed.value().isEmpty()){
+            metricName = classLevelTimed.value();
+        }
+        else {
+            metricName = TimedAspect.DEFAULT_METRIC_NAME;
+        }
+        Builder builder = Timer.builder(timedMetricNameResolver.apply(metricName, invocation));
+        if (methodLevelTimed != null) {
+            builder.description(methodLevelTimed.description().isEmpty() ? null : methodLevelTimed.description())
+                .publishPercentileHistogram(methodLevelTimed.histogram())
+                .publishPercentiles(methodLevelTimed.percentiles().length == 0 ? null : methodLevelTimed.percentiles());
+        }
+        else {
+            builder.description(classLevelTimed.description().isEmpty() ? null : classLevelTimed.description())
+                .publishPercentileHistogram(classLevelTimed.histogram())
+                .publishPercentiles(classLevelTimed.percentiles().length == 0 ? null : classLevelTimed.percentiles());
+        }
+
+        if (classLevelTimed != null) {
+            builder.tags(classLevelTimed.extraTags());
+        }
+        if (methodLevelTimed != null) {
+            builder.tags(methodLevelTimed.extraTags());
+        }
+        builder.tags(timedTagsResolver.apply(invocation));
```

Comment:
Currently, if method-level annotation has an extra tag with name `method`, it will be `added` to the list of tags, producing 2 tags with the same name, which leads to an error on `influxdb` side. Is it possible to merge the tags in a way that method-level tags will `override` the tags defined by `TimedMethodInterceptor`? It will probably require a change in type of io.micrometer.core.instrument.Timer.Builder#tags variable from `List` to `Set`.

Reply:
I can use set here to ensure tags uniqueness, but I think it's better to change builder and/or `Meter.Id` where all builders supply tags. 
I think problem is in `Meter.Id`:
```
this.tags = Collections.unmodifiableList(stream(tags.spliterator(), false)
                .sorted(Comparator.comparing(Tag::getKey))
                .distinct()
                .collect(Collectors.toList()));
```
here `distinct()` checks both key and value, not only key. @jkschneider what do you think?